docs/UpgradeGuide2.md

```diff
@@ -0,0 +1,543 @@
+# Upgrading ddtrace from 1.x to 2.0
+
+Upgrading `ddtrace` from 1.x to 2.0 introduces some breaking changes which are outlined below.
+
+[**Basic Usage**](#2.0-basic-usage)
+
+In most cases, (e.g. when you use out-of-the-box instrumentation and a configuration file) only minor changes to your current setup are needed: most applications take just minutes to update.
+
+- [Requires Ruby 2.5+](#2.0-requires-ruby-2.5+)
+- [Extracts datadog-ci gem](#2.0-extracts-ci)
+- [Configuration changes](#2.0-configuration-changes)
+  - [Replace `use` with `instrument`](#2.0-use-instrument)
+  - [Type checking](#2.0-type-checking)
+  - [Propagation default](#2.0-propagation-default)
+  - [Options](#2.0-options)
+
+---
+
+[**Advanced Usage**](#2.0-advanced-usage)
+
+If you have custom instrumentation, sampling or processing behavior, additional changes may be required. See the following sections for details about what features changed and how to use them.
+
+- [Frozen String Literal](#2.0-frozen-string)
+- [Tracing API](#2.0-tracing-api)
+- [Configuration API](#2.0-configuration-api)
+- [Log Correlation](#2.0-log-correlation)
+- [Distributed Tracing](#2.0-distributed-tracing)
+- [Transport](#2.0-transport)
+- [Sampling](#2.0-sampling)
+
+---
+
+[**Upgrade Instrumentations**](#2.0-upgrade-instrumentations)
+
+Outlines the changes for our instrumentations.
+
+- [Error Handling](#2.0-error-handling)
+- [Option `error_status_codes`](#2.0-error-status-codes)
+- [List of Integration changes](#2.0-list-of-integration-changes)
+
+<h2 id="2.0-basic-usage">Basic Usage</h2>
+
+<h3 id="2.0-requires-ruby-2.5+">Requires Ruby 2.5+</h3>
+
+The minimum Ruby version requirement for ddtrace 2.x is 2.5.0. For prior Ruby versions, you can use ddtrace 1.x. see more with our support policy.
+
+<h3 id="2.0-extracts-ci">Extracts datadog-ci gem</h3>
+
+The CI visibility component has been extracted as a separate gem named [datadog-ci](https://github.com/DataDog/datadog-ci-rb), and will no longer be installed together with ddtrace.
+
+If you are using our CI visibility product, include `datadog-ci` in your Gemfile and learn more about the [setup](https://github.com/DataDog/datadog-ci-rb).
+
+```ruby
+gem 'ddtrace', '>= 2'
+gem 'datadog-ci'
```

Comment:
minor: I would recommend putting datadog-ci under test group so that it does not load in production:

```ruby
gem 'ddtrace', '>= 2'

group :test do
  gem 'datadog-ci'
end
```